### PR TITLE
[open62541] Update to v1.0

### LINKED ISSUE #9766: [open62541] update to 1.0
Library name: open62541 ( https://open62541.org/ )

New version number: 1.0

Other information that may be useful (release notes, etc...):
* https://github.com/open62541/open62541/releases/tag/v1.0

### PR DESCRIPTION
**Pull request by yurybura**
[open62541] Update to v1.0

Fixes #9766

### CODE REVIEW

**Comment by traversaro** [comment]
Thanks @yurybura for working on this. 

As the dependency on python six was dropped in https://github.com/iit-danieli-joint-lab/idjl-software/pull/725 , I guess we can now also remove all the code necessary to install `six`, see https://github.com/microsoft/vcpkg/pull/9767/files#diff-d7c83a3acc5f48906322217e14675fe7L36 .

**Comment by yurybura** [comment]
Hi @traversaro
Thanks for comment. Unfortunately your link is unavailable. I'll try to remove python-six after my investigations.

**Comment by traversaro** [comment]
> 
> 
> Hi @traversaro
> Thanks for comment. Unfortunately your link is unavailable. I'll try to remove python-six after my investigations.

Sorry, I had copy pasted the wrong comment. The correct one is https://github.com/open62541/open62541/pull/2493 .

**Comment by yurybura** [comment]
Requires #9752 merge to fix Windows CI...

File: ports/open62541/portfile.cmake [hunk 1]
```diff
@@ -1,77 +1,50 @@
...
+)
 file(WRITE ${SOURCE_PATH}/CMakeLists.txt "${OPEN62541_CMAKELISTS}")
 
-if(CMAKE_HOST_WIN32)
-    set(EXECUTABLE_SUFFIX ".exe")
-else()
-    set(EXECUTABLE_SUFFIX "")
-endif()
-
 vcpkg_find_acquire_program(PYTHON3)
```
Comment (cenit) [inline ports/open62541/portfile.cmake]: Also this part should not be required anymore, right?

Comment (yurybura) [reply ports/open62541/portfile.cmake]: No, Python is still required to build.
CMakeLists.txt contains:
`find_package(PythonInterp REQUIRED)`

File: ports/open62541/CONTROL [hunk 2]
```diff
@@ -1,3 +1,3 @@
 Source: open62541
-Version: 0.3.0-4
+Version: 1.0
 Description: open62541 is an open source C (C99) implementation of OPC UA licensed under the Mozilla Public License v2.0.
```
Comment (NancyLi1013) [inline ports/open62541/CONTROL]: Could you please add homepage here?

File: ports/open62541/portfile.cmake [hunk 3]
```diff
@@ -1,77 +1,42 @@
 include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/open62541/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove this line?

File: ports/open62541/portfile.cmake [hunk 4]
```diff
@@ -1,77 +1,42 @@
...
 
-if(NOT CMAKE_HOST_WIN32)
-    message("${PORT} currently requires the following tools from the system package manager:\n    python-six\n\nThis can be installed on Ubuntu systems via apt-get install python-six python3-six (depending on your current python default interpreter)")
-endif()
-
 if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
     message(FATAL_ERROR "${PORT} does not currently support UWP")
 endif()
```
Comment (NancyLi1013) [inline ports/open62541/portfile.cmake]: This can be updated as` vcpkg_fail_port_install(ON_TARGET "UWP")`.

File: ports/open62541/portfile.cmake [hunk 5]
```diff
@@ -1,77 +1,42 @@
...
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/share/open62541/tools)
 
-vcpkg_copy_pdbs()
-
 file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+
+# CMake integration test
+vcpkg_test_cmake(PACKAGE_NAME ${PORT})
```
Comment (NancyLi1013) [inline ports/open62541/portfile.cmake]: `vcpkg_test_cmake()` is deprecated.
Could you remove it?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/open62541/CONTROL (hunk 2 above)
Comment (yurybura) [reply ports/open62541/CONTROL]: Added.

File: ports/open62541/portfile.cmake (hunk 3 above)
Comment (yurybura) [reply ports/open62541/portfile.cmake]: Fixed.

File: ports/open62541/portfile.cmake (hunk 4 above)
Comment (yurybura) [reply ports/open62541/portfile.cmake]: Fixed.

File: ports/open62541/portfile.cmake (hunk 5 above)
Comment (yurybura) [reply ports/open62541/portfile.cmake]: Removed.

**Comment by yurybura** [comment]
@NancyLi1013 thank you for review! I've fixed issues.

**Review by NancyLi1013 — APPROVED** [review]

File: ports/open62541/portfile.cmake [hunk 6]
```diff
@@ -1,77 +1,35 @@
-include(vcpkg_common_functions)
+vcpkg_fail_port_install(ON_TARGET "UWP")
 
-if(NOT CMAKE_HOST_WIN32)
-    message("${PORT} currently requires the following tools from the system package manager:\n    python-six\n\nThis can be installed on Ubuntu systems via apt-get install python-six python3-six (depending on your current python default interpreter)")
```
Comment (JackBoosY) [inline ports/open62541/portfile.cmake]: Can you be sure again that the port does not depend on `python-six` and `python3-six` now?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/open62541/portfile.cmake (hunk 6 above)
Comment (yurybura) [reply ports/open62541/portfile.cmake]: According to open62541/open62541#2493 dependency to python-six was removed 6 months ago. Anyway I'll check in the sources.

Comment (yurybura) [reply ports/open62541/portfile.cmake]: I can confirm that python-six dependency was removed in v1.0.

**Review by JackBoosY — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks @yurybura for the PR and everyone that helped review it!

**Pull request merged**